File: app/src/main/java/torille/fi/lurkforreddit/di/modules/AppModule.java
package torille.fi.lurkforreddit.di.modules;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by eva on 22.4.2017.
 */
@Module
public final class AppModule {

    private final Application mApplication;

    public AppModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return mApplication;
    }
}

File: app/src/main/java/torille/fi/lurkforreddit/di/modules/AppModule.kt
package torille.fi.lurkforreddit.di.modules

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Dagger 2 module providing Application
 */
@Module
class AppModule(private val application: Application) {

    @Provides
    @Singleton
    internal fun bindContext(): Context {
        return application.applicationContext
    }
}
